engine/src/main/java/org/hibernate/validator/resourceloading/PlatformResourceBundleLocator.java

```diff
@@ -164,21 +166,42 @@ private ResourceBundle loadBundle(ClassLoader classLoader, Locale locale, String
 	}
 
 	/**
+	 * Check whether ResourceBundle.Control is available, which is needed for bundle aggregation. If not, we'll skip
+	 * resource aggregation.
+	 * <p>
+	 * It is *not* available
+	 * <ul>
+	 * <li>in the Google App Engine environment</li>
+	 * <li>when running HV as Java 9 named module (which would be the case when adding a module-info descriptor to the
+	 * HV JAR)</li>
+	 * </ul>
 	 *
-	 * In an Google App Engine environment bundle aggregation is not possible, since ResourceBundle.Control
-	 * is not on the list of white listed classes in this environment.
-	 * to create AggregateResourceBundle.CONTROL proactively, if it fails skip resource aggregation.
-	 *
-	 * @see <a href="http://code.google.com/appengine/docs/java/jrewhitelist.html">JRE whitelist</a>
+	 * @see <a href="http://code.google.com/appengine/docs/java/jrewhitelist.html">GAE JRE whitelist</a>
 	 * @see <a href="https://hibernate.atlassian.net/browse/HV-1023">HV-1023</a>
+	 * @see <a href="http://download.java.net/java/jdk9/docs/api/java/util/ResourceBundle.Control.html>ResourceBundle.Control</a>
 	 */
 	private static boolean determineAvailabilityOfResourceBundleControl() {
 		try {
-			@SuppressWarnings("unused")
 			ResourceBundle.Control dummyControl = AggregateResourceBundle.CONTROL;
-			return true;
+
+			if ( dummyControl == null ) {
+				return false;
+			}
+
+			Method getModule = run( GetMethod.action( Class.class, "getModule" ) );
+			// not on Java 9
+			if ( getModule == null ) {
+				return true;
+			}
+
+			// on Java 9, check whether HV is a named module
+			Object module = getModule.invoke( PlatformResourceBundleLocator.class );
+			Method isNamedMethod = run( GetMethod.action( module.getClass(), "isNamed" ) );
+			boolean isNamed = (Boolean) isNamedMethod.invoke( module );
```

Comment:
I tested this manually, which I think - while not perfect - is good enough. And even if things went wrong, the impact is low, so I think that's ok.